step-capstone/src/scripts/Optimization.js

```diff
@@ -0,0 +1,85 @@
+const BREAKFAST_TIMERANGE = [6, 10];
+const LUNCH_TIMERANGE = [11, 15];
+const DINNER_TIMERANGE = [17, 21];
+
+export const getOptimalRoute = function (travelObjects, origin, destination) {
+    let request = {
+        origin: new window.google.maps.LatLng(origin.coordinates.lat, origin.coordinates.lng),
+        destination: new window.google.maps.LatLng(destination.coordinates.lat, destination.coordinates.lng),
+        waypoints: travelObjects.map(travelObject => {
+            return { location: new window.google.maps.LatLng(travelObject.coordinates.lat, travelObject.coordinates.lng), stopover: true }
+        }),
+        travelMode: "DRIVING",
+        optimizeWaypoints: true
+    }
+    var directionsService = new window.google.maps.DirectionsService();
+
+    return new Promise(res => {
+        directionsService.route(request, function (result, status) {
+            if (status === "OK") {
```

Comment:
What if the status here is not "OK"? It would be good to add in some error-handling here or have a multiple retry loop to handle flaky random errors from the directions service.

Reply:
added error handling